tests/perf/benchmark.py

```diff
@@ -308,10 +292,84 @@ def run(
+            raw_data = pd.read_csv(csv_file)
+            if (
+                "train/epoch" in raw_data.columns  # check it's csv of train result
+                and all(  # check meta info is same
+                    str(raw_data.iloc[0].get(key, "NOT_IN_CSV")) == tags.get(key, "NOT_IN_TAG")
+                    for key in ["data_group", "data", "model", "task", "seed"]
+                )
+            ):
+                return csv_file.parent
+        return None
+
+    def _copy_prev_train_dir(self, prev_train_dir: Path, work_dir: Path) -> Path:
+        work_dir.mkdir(parents=True, exist_ok=True)
+        new_train_dir = work_dir / prev_train_dir.name
+        shutil.copytree(prev_train_dir, new_train_dir, ignore_dangling_symlinks=True)
+        cache_dir = work_dir / ".latest" / "train"
+        cache_dir.parent.mkdir(exist_ok=True)
+        cache_dir.symlink_to(Path("..") / new_train_dir.relative_to(work_dir))
+
+        return new_train_dir
+
+    def _run_test(
+        self,
+        work_dir: Path | str,
+        dataset: Dataset,
+        tags: dict[str, str],
+        criteria: list[Criterion],
+        checkpoint: Path | str | None = None,
+        what2test: Literal["train", "export", "optimize"] = "train",
+    ) -> None:
+        """Run otx test and update result csv file to align it's indices to the current task."""
+        replace_map = {
+            "train": {"test_": "test/", "{pre}": "export/"},
+            "export": {"test": "export", "{pre}": "export/"},
+            "optimize": {"test": "optimize", "{pre}": "optimize/"},
+        }
+
+        command = [
+            "otx",
+            "test",
+            "--work_dir",
+            str(work_dir),
+        ]
+        if checkpoint is not None:
+            command.extend(["--checkpoint", str(checkpoint)])
+        for key, value in dataset.extra_overrides.get("test", {}).items():
+            command.append(f"--{key}")
+            command.append(str(value))
+
+        start_time = time()
+        self._run_command(command)
+        extra_metrics = {f"test({what2test})/e2e_time": time() - start_time}
+
+        self._rename_raw_data(
+            work_dir=work_dir / ".latest" / "test",
+            replaces=replace_map[what2test],
+        )
+        self._log_metrics(
+            work_dir=work_dir / ".latest" / "test",
+            tags=tags,
+            criteria=criteria,
+            extra_metrics=extra_metrics,
+        )
+
+    def _run_command(self, command: list[str], **kwargs) -> None:
+        """Run command using 'subprocess.run'.
+
+        Args:
+            command (list[str]): command to execute.
+            kwags: arguments to 'subprocess.run'.
+        """
         print(" ".join(command))
+        kwargs["check"] = True
         if not self.dry_run:
-            subprocess.run(command, check=True)  # noqa: S603
+            subprocess.run(command, **kwargs)  # noqa: S603, PLW1510
```

Comment:
Needed for pre-commit issue?

Reply:
I want to pass additional arguments for `subprocess.run` and this is for dealing with a case that user passes `check` in kwargs.